### ISSUE TITLE
Way to Go From Method Implementation to Interface Method Declaration

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#14588]
From https://github.com/Microsoft/vscode/issues/22385

**TypeScript Version:**  2.2.1

**Code**

```ts
interface ITest {
    test(): void;
}

class Test implements ITest {
    test() {
        return null;
    }
}
```

**Request**
Running `go to definition` on `test` in class `Test` currently does not return `ITest::test`. Neither does `go to type definition`.

In VSCode, we would like some way to jump from a class's interface method implementation  or abstract method implementation to the interface method declaration or abstract method declaration  itself. I'm not sure which command makes sense to do this, either `go to definition` or `go to type definition`, but we are open to any suggestions

**Comment by RyanCavanaugh** [comment]
FWIW the "Find All References" command already does this. I think what you essentially want here is "Find all referencing declarations", which would be trivial to implement.

**Comment by pkeuter** [comment]
What I think would be a nice addition to vscode is that you could jump from the implementation of an interface to the Interface method itself. There are two options now: "go to definition" and "go to type definition", and they both give a "not found" error. Which, in my opinion, could be better.

**Comment by mhegazy** [comment]
i do not think this is a **definition**. what about a class that overrides a declaration? what about an object literal that really defines this function?

i think go to type seems more reasonable here.

**Comment by mjbvz** [comment]
VS Code now has a `go to declaration` feature. While I believe it was originally added for c++, it may make sense for this use case in TS
